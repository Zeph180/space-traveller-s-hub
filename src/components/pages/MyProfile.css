.container {
  display: flex;
  flex-direction:"row";
  width: 80%;
  margin-left: 10%;
  margin-top: var(--tenPx);
  justify-content: space-between;
}

.container > * {
  width: 48%;
}

.my-rocket, .my-mission {
  border: 1px solid;
  padding: 20px;
  margin: var(--zero);
}

.toper :nth-child(1) {
  border-top-right-radius: var(--tenPx);
  border-top-left-radius: var(--tenPx);
}

.cat-headers {
  width: 100%;
  font-size: 25px;
  padding: var(--tenPx);
}
