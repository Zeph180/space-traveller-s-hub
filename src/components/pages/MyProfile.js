/* eslint-disable react/jsx-key */
import { useSelector } from 'react-redux';

export default function MyProfile() {
  const { missions } = useSelector((state) => state.missions);
  const { rockets } = useSelector((state) => state.rockets);
  return (
    <>
      <p>My profile</p>
      <div>
        {missions.map((mission) => (mission.joined ? <p>{mission.mission_name}</p> : null))}
      </div>
      <div>
        {rockets.map((rocket) => (rocket.reserved ? <p>{rocket.name}</p> : null))}
      </div>
    </>
  );
}
