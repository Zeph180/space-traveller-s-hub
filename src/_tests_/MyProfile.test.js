import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import MyProfile from '../components/pages/MyProfile';

describe('MyProfile', () => {
  const initialState = {
    missions: {
      missions: [
        { mission_name: 'Mission 1', joined: true },
        { mission_name: 'Mission 2', joined: false },
        { mission_name: 'Mission 3', joined: true },
      ],
    },
    rockets: {
      rockets: [
        { name: 'Rocket 1', reserved: true },
        { name: 'Rocket 2', reserved: false },
        { name: 'Rocket 3', reserved: true },
      ],
    },
  };
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('should render correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
