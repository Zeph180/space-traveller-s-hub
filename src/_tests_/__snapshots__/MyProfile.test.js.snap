// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile should render correctly 1`] = `
<section
  class="container"
>
  <div>
    <h3
      class="cat-headers"
    >
      My Missions
    </h3>
    <div
      class="toper"
    >
      <p
        class="my-mission"
      >
        Mission 1
      </p>
      <p
        class="my-mission"
      >
        Mission 3
      </p>
    </div>
  </div>
  <div>
    <h3
      class="cat-headers"
    >
      My Rockets
    </h3>
    <div
      class="toper"
    >
      <p
        class="my-rocket"
      >
        Rocket 1
      </p>
      <p
        class="my-rocket"
      >
        Rocket 3
      </p>
    </div>
  </div>
</section>
`;
